ascii/ascii_test.go

```diff
@@ -31,6 +32,274 @@ func TestControlCodes(t *testing.T) {
 	}
 }
 
+func PrettyFormat(ircText string) string {
+	out := strings.ReplaceAll(ircText, "\x03", "\x1b[033m[C]\x1b[0m")
+	out = strings.ReplaceAll(out, "\x02", "\x1b[034m[B]\x1b[0m")
+	out = strings.ReplaceAll(out, "\x1F", "\x1b[035m[U]\x1b[0m")
+	out = strings.ReplaceAll(out, "\x1D", "\x1b[036m[I]\x1b[0m")
+	return out
+}
+
+func PrintExpectedVsParsed(expected string, ascii *ASCII, t *testing.T) {
+	t.Errorf(" Input   : %s", PrettyFormat(expected))
+	t.Errorf(" Parsed  : %s", PrettyFormat(string(ascii.Bytes())))
+}
+
+func TestColorCodes(t *testing.T) {
+	s := "hello \x0315,01hello"
+	//    012345         6789
+	a, _ := NewASCII(s)
+	if c := a.Get(2, 0); c == nil || c.Value != 'l' {
```

Comment:
Too much repetition-- idiomatic go uses a table driven test pattern like https://github.com/chzchzchz/sitbot/blob/master/msl/grammar_test.go#L49

Struct would be like
```go
struct {
input string
col []int
expected []Cell
}
```

Driver would be like
```go
for _, tt := range tests {
a, _ := NewAscii(tt.input);
for i := range tt.col {
c := a.Get(tt.col[i], 0)
if c == nil || c != tt.expected[i] {
t.Errorf("got %+v, want %+v at %d for %q",  c, tt.expected[i], tt.col[i], tt.input)
}
}
```